File: ORB.Core/Models/CorporateRisk.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class CorporateRisk : Risk
    {
        public CorporateRisk() : base()
        {
            ChildRisks = new HashSet<CorporateRisk>();
            RiskRiskTypes = new HashSet<RiskRiskType>();
            RiskMitigationActions = new HashSet<CorporateRiskMitigationAction>();
            CorporateRiskRiskMitigationActions = new HashSet<CorporateRiskRiskMitigationAction>();
            RiskUpdates = new HashSet<CorporateRiskUpdate>();
        }

        public int? DepartmentalObjectiveID { get; set; }

        public virtual DepartmentalObjective DepartmentalObjective { get; set; }
        public virtual CorporateRisk LinkedRisk { get; set; }
        public virtual ICollection<CorporateRisk> ChildRisks { get; set; }
        public virtual ICollection<RiskRiskType> RiskRiskTypes { get; set; }
        public virtual ICollection<CorporateRiskMitigationAction> RiskMitigationActions { get; set; }
        public virtual ICollection<CorporateRiskRiskMitigationAction> CorporateRiskRiskMitigationActions { get; set; }
        public virtual ICollection<CorporateRiskUpdate> RiskUpdates { get; set; }
    }
}

File: corporate-reporting/src/types/CorporateRisk.ts
import { ICorporateRiskMitigationAction } from "./CorporateRiskMitigationAction";
import { ICorporateRiskUpdate } from "./CorporateRiskUpdate";
import { IRisk, Risk } from "./Risk";

export interface ICorporateRisk extends IRisk {
    DepartmentalObjectiveID: number;
    RiskUpdates?: ICorporateRiskUpdate[];
    RiskMitigationActions?: ICorporateRiskMitigationAction[];
}

export class CorporateRisk extends Risk implements ICorporateRisk {
    public DepartmentalObjectiveID = null;
}
